theano/gof/link.py

```diff
@@ -504,11 +507,13 @@ def map_storage(fgraph, order, input_storage, output_storage):
 #             raise TypeError("Cannot link a graph with non-constant orphans.", orphan)
 #         storage_map[orphan] = [orphan.data]
 
+    # allocate output storage
     if output_storage is not None:
```

Comment:
Can you test this case:

check that it input_storage and storage_map is provided or output_storage, assert that they are the same? If tests pass like that, I think we should keep the assert. This make the understanding easier I think.